h2o-parsers/h2o-parquet-parser/src/test/java/water/parser/parquet/ParseTestParquet.java

```diff
@@ -213,6 +217,35 @@ public void check(Frame f) {
     assertFrameAssertion(assertion);
   }
 
+  @Test
+  public void  testParseLargeStringBlock() {
```

Comment:
This test requires approx. -Xmx20g to run, as the very first allocation takes x 4 the space, which already hits the MAX_STR_LEN boundary.